src/js/utils/obj.js

```diff
@@ -30,3 +31,55 @@ export function reduce(object, fn, initial = 0) {
     (accum, key) => fn(accum, object[key], key),
     initial);
 }
+
+/**
+ * Object.assign-style object shallow merge/extend.
+ *
+ * @param  {Object} target
+ * @param  {Object} ...sources
+ * @return {Object}
+ */
+export function assign(target, ...sources) {
```

Comment:
FWIW, I use a similar helper extensively at work (Redux app) and we went ahead and defaulted to setting the target to an empty object so the result is always a shallow copy. Not sure if that makes sense here, though.